### PR TITLE
Add route for creating streams

### PR DESCRIPTION
**Pull request by mlejva**
This PR adds:
- API route that creates a stream for given sandbox ID  `POST /stream/sandbox`
- Dynamic page for displaying the livestream video `/stream/sandbox/[sandboxId]

### CODE REVIEW

File: apps/web/src/app/(dashboard)/dashboard/layout.tsx [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
-
 export default async function Layout({ children }) {
   return (
-    <LayoutDashboard>
-      {children}
-      <Toaster />
-    </LayoutDashboard>
+    <div className="h-full w-full">
+      <main className="w-full h-full flex flex-col">
```
Comment (mishushakov) [inline apps/web/src/app/(dashboard)/dashboard/layout.tsx:7]: Why didn't you update LayoutDashboard instead?

File: apps/web/src/app/api/stream/sandbox/route.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+export async function POST(request: Request) {
+  // TODO: UseG apiKey
+  // const apiKey = request.headers.get('X-API-Key')
+  const { sandboxId } = await request.json()
+  console.log('sandboxId', sandboxId)
+
+  // if (!apiKey) {
+  //   return NextResponse.json({ error: 'Missing X-API-Key' }, { status: 400 })
+  // }
```
Comment (mishushakov) [inline apps/web/src/app/api/stream/sandbox/route.ts]: There's no security?

File: apps/web/src/app/layout.tsx [hunk 3]
```diff
@@ -39,8 +37,18 @@ declare global {
...
+  const isSandboxRoute = params?.sandboxId !== undefined
+
+  if (isSandboxRoute) {
+    return (
+      <html lang="en">
+        <head></head>
+        <body>{children}</body>
+      </html>
+    )
+  }
```
Comment (mishushakov) [inline apps/web/src/app/layout.tsx]: Is this the correct way to do it?

File: apps/web/src/app/stream/sandbox/[sandboxId]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  if (error || !data) {
+    return null
+  }
+
+  return { sandboxId, playbackId: data.playback_id }
+}
+
+export default async function StreamPage({ params }: { params: { sandboxId: string } }) {
+  const stream = await fetchStream(params.sandboxId)
```
Comment (mishushakov) [inline apps/web/src/app/stream/sandbox/[sandboxId]/page.tsx]: No security?

File: apps/web/src/components/ui/Popover.tsx [hunk 5]
```diff
@@ -16,7 +16,7 @@ const PopoverContent = React.forwardRef<
   <PopoverPrimitive.Portal>
     <PopoverPrimitive.Content
       ref={ref}
-      align={align}
+      align={align as any}
```
Comment (mishushakov) [inline apps/web/src/components/ui/Popover.tsx]: Avoid "any"

File: apps/web/src/middleware.ts [hunk 6]
```diff
@@ -5,6 +5,7 @@ export async function middleware(req: NextRequest): Promise<NextResponse> {
   if (req.method !== 'GET') return NextResponse.next()
 
   const url = new URL(req.nextUrl.toString())
+  console.log('url', url)
```
Comment (mishushakov) [inline apps/web/src/middleware.ts]: Remove this

File: apps/web/src/types/mux-player-react.d.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+declare module '@mux/mux-player-react' {
+  export interface MuxPlayerProps {
+    [key: string]: any;
+    playbackId: string;
+    metadata?: Record<string, any>;
+  }
+
+  export default function MuxPlayer(props: MuxPlayerProps): JSX.Element
+}
```
Comment (mishushakov) [inline apps/web/src/types/mux-player-react.d.ts]: Why is this required? They don't have typings?

**Review by mishushakov — COMMENTED** [review]
Added feedback

File: apps/web/src/app/api/stream/sandbox/route.ts (hunk 2 above)
Comment (mishushakov) [reply apps/web/src/app/api/stream/sandbox/route.ts]: Also, why are we using api routes, our backend is in Go

File: apps/web/src/app/(dashboard)/dashboard/layout.tsx (hunk 1 above)
Comment (mlejva) [reply apps/web/src/app/(dashboard)/dashboard/layout.tsx:7]: I deleted the component and moved the layout directly to this `layout.tsx` file. The `LayoutDashboard` wasn't used anywhere but here and there wasn't much logic in it.

File: apps/web/src/app/layout.tsx (hunk 3 above)
Comment (mlejva) [reply apps/web/src/app/layout.tsx]: Shouldn't be there. I forgot to delete. Thanks for catching.

File: apps/web/src/types/mux-player-react.d.ts (hunk 7 above)
Comment (mlejva) [reply apps/web/src/types/mux-player-react.d.ts]: Correct. That was my experience. I installed their packages and the linter couldn't find declarations. They also don't seem to have `@types/...` package.

File: apps/web/src/app/api/stream/sandbox/route.ts (hunk 2 above)
Comment (mlejva) [reply apps/web/src/app/api/stream/sandbox/route.ts]: Discussed this @ValentaTomas. We felt like this is a better place for the implementation given it's right next to the actual page rendering the stream.

File: apps/web/src/app/stream/sandbox/[sandboxId]/page.tsx (hunk 4 above)
Comment (mlejva) [reply apps/web/src/app/stream/sandbox/[sandboxId]/page.tsx]: Discussed this with @ValentaTomas. Streams are public so reading them should be public. Maybe we could allow creating them only if you're sending an E2B API key @ValentaTomas ?

File: apps/web/src/types/mux-player-react.d.ts (hunk 7 above)
Comment (mlejva) [reply apps/web/src/types/mux-player-react.d.ts]: However, [their project](https://github.com/muxinc/elements/tree/main/packages/mux-player-react) is in TypeScript. It's weird that the linter is failing on it because it doesn't fail on any other package.

Comment (mlejva) [reply apps/web/src/types/mux-player-react.d.ts]: Resolved. Seemed to be a cache issue.

**Comment by mlejva** [comment]
@ValentaTomas I also realized - we should add a migration for the new Supabase table. Right?

File: apps/web/src/app/stream/sandbox/[sandboxId]/page.tsx (hunk 4 above)
Comment (ValentaTomas) [reply apps/web/src/app/stream/sandbox/[sandboxId]/page.tsx]: Creating them is behind API key+sandbox ID.

For access, we want there to be just a link you can follow. The most straightforward solution here then could be that we also pass some per sandbox stream secret when we create the sandbox that is then added to the link returned in SDK in query params.

That could offer more flexibility in the future, because right now the sandbox ID also serve as the secret you use for connecting.

Comment (ValentaTomas) [reply apps/web/src/app/stream/sandbox/[sandboxId]/page.tsx]: To future-proof this, it might be worth adding and passing such a field so we don't face compatibility problems later.

**Comment by mlejva** [comment]
I added additional security layer with `token` URL parameter
Two API routes:

## 1. `POST /api/stream/sandbox`
Creates a new stream for a sandbox if a stream doesn't exist yet (each sandbox can have exactly one stream at any given time now). Returns the token and stream key
- When creating a new stream, the token gets generated in the DB via a postgres function

## 2. `GET /api/stream/sandbox/[sandboxId]`
Checks the DB for existing stream for the given sandbox and returns the token for this stream

File: apps/web/src/app/stream/sandbox/[sandboxId]/page.tsx (hunk 4 above)
Comment (mlejva) [reply apps/web/src/app/stream/sandbox/[sandboxId]/page.tsx]: Done

**Review by ValentaTomas — APPROVED** [review]
